advanced_filters/forms.py

```diff
@@ -361,7 +361,7 @@ def initialize_form(self, instance, model, data=None, extra=None):
                     AdvancedFilterQueryForm._parse_query_dict(
                         field_data, model))
 
-        formset = AFQFormSetNoExtra if extra is None else AFQFormSet
+        formset = AFQFormSetNoExtra if not extra or not instance else AFQFormSet
```

Comment:
You're right about `not extra`, good catch!

However, I'm not sure if I like `or not instance` here. If we're editing a new form and get an error on submit,  the instance is still None, but I think the user expects to see an extra form row (if it was there before the error).